ddtrace/contrib/django/utils.py

```diff
@@ -88,6 +88,16 @@ def get_request_uri(request):
 
     # Build request url from the information available
     # DEV: We are explicitly omitting query strings since they may contain sensitive information
-    return parse.urlunparse(
-        parse.ParseResult(scheme=request.scheme, netloc=host, path=request.path, params="", query="", fragment="",)
-    )
+    urlparts = dict(scheme=request.scheme, netloc=host, path=request.path)
+    urlparts_omit = ["params", "query", "fragment"]
+
+    # DEV: If all url parts are byte strings then return uri should be a byte
+    # string, otherwise cast all parts as strings to avoid mixing bytes and
+    # string as arguments
+    if all(isinstance(val, bytes) or val is None for val in urlparts.values()):
```

Comment:
I feel like this could be optimized a bit, seems we are potentially iterating over items a few times... but maybe we can't because we need to know what default to use in the omit?

Reply:
I'm not sure if we can avoid iterating over the three items in `urlparts` since we need to determine first whether any are strings in order to cast all of them.

Reply:
The error seems to be based on the type of the first argument: 

https://github.com/python/cpython/blob/02d126aa09d96d03dcf9c5b51c858ce5ef386601/Lib/urllib/parse.py#L491-L492

https://github.com/python/cpython/blob/02d126aa09d96d03dcf9c5b51c858ce5ef386601/Lib/urllib/parse.py#L111-L125